server/pull_request_handler.go

```diff
@@ -60,7 +60,8 @@ func (s *Server) pullRequestEventHandler(w http.ResponseWriter, r *http.Request)
 			mlog.Debug("Triggered CircleCI", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("fullname", pr.FullName))
 		}
 
-		s.addHacktoberfestLabel(ctx, pr)
+		go s.postPRWelcomeMessage(ctx, pr)
+		go s.addHacktoberfestLabel(ctx, pr)
```

Comment:
Any reason these are spun in goroutines? Unless they are long-running tasks, it would be better to just run them sequentially.

Reply:
No they are not long running tasks. Are there downsides of using a go routine?

Reply:
Randomly spinning up goroutines is never a good idea. Whenever we spin up a goroutine, we need to think of how will it be shut down, what's the cancelling mechanism, so on and so forth.

An immediate bug here if we would create goroutines is the context. Since the same context is passed to a goroutine, it will get cancelled the moment it exits the handler. And therefore, it will cause all actions performed in the goroutine to be cancelled. This would need new contexts to be set, which loses out on the original point of being able to set the correct timeouts.

So more things go out of control as we randomly spin up goroutines. The proper way is to have a job queue to handle the entire request asynchronously. But that's a long, long way ahead.

Reply:
Context is a good point. I haven't thought about that one.

Reverted my changes.